backend/api/serializers/HeatDisplaySerializer.py

```diff
@@ -21,47 +21,50 @@ def get_athlete_full_name(self, instance):
                 return athlete.full_name
         return None
 
+
 class LaneSerializer(serializers.ModelSerializer):
     athlete_full_name = serializers.SerializerMethodField()
     # Setting write_only directly on the field as HeatDurationField is not a model field
     heat_time = HeatDurationField()
 
     class Meta:
         model = Heat
-        fields = ('id', 'num_heat', 'athlete', 'athlete_full_name', 'heat_time')
+        fields = ('id', 'num_heat', 'athlete',
+                  'athlete_full_name', 'heat_time')
 
         extra_kwargs = {
             'athlete': {'read_only': True},
             'athlete_full_name': {'read_only': True},
         }
 
-
     def get_athlete_full_name(self, instance):
         if isinstance(instance, Heat):
             athlete = instance.athlete
             if athlete:
                 return athlete.full_name
         return None
-    
+
     @transaction.atomic
     def update(self, instance, validated_data):
-        #Updates heat_time
-        instance.heat_time = validated_data.get('heat_time')
-        instance.save()
-    
-        #Register the time on TimeRecord
-        athlete = validated_data.get('athlete', instance.athlete)
+        # Register the time on TimeRecord
         event_instance = self.context['event']
         event_type = event_instance.event_type
         swim_meet = event_instance.swim_meet
-        TimeRecord.objects.create(
-            athlete = athlete,
-            event_type = event_type,
-            swim_meet = swim_meet,
-            date = swim_meet.date,
-            time = instance.heat_time
+
+        time_record, created = TimeRecord.objects.update_or_create(
```

Comment:
Could you please remove time_record and created, please, because we are not using them.